### ISSUE TITLE
taskResult element only included if an exception occurred (output consistency)

### ISSUE THREAD

**Issue by bitsgalore** [veraPDF/veraPDF-library#1336]
I'm running VeraPDF (extraction only) on [this file](https://github.com/openpreserve/format-corpus/blob/master/pdfCabinetOfHorrors/corruptionOneByteMissing.pdf?raw=true) using the command below:

```
verapdf --off --extract corruptionOneByteMissing.pdf > corruptionOneByteMissing.xml
```

This PDF is corrupted, which causes an exception that is reported in the [output file's](https://github.com/KBNLresearch/pdf-characterisation/blob/main/output/horror/corruptionOneByteMissing-vera.xml) *taskResult* element (as expected):

```xml
<taskResult type="PARSE" isExecuted="true" isSuccess="false">
    <duration start="1683818745078" finish="1683818745121">00:00:00.043</duration>
    <exceptionMessage>Exception: Couldn't parse stream caused by exception: Pages not found</exceptionMessage>
</taskResult>
```
However, when I run the same command on a PDF that can be parsed without exceptions, the *taskResult* element is not included in the output at all. See for example [this PDF](https://github.com/openpreserve/format-corpus/blob/master/pdfCabinetOfHorrors/embedded_video_avi.pdf?raw=true), which results in [this output](https://github.com/KBNLresearch/pdf-characterisation/blob/main/output/horror/embedded_video_avi-vera.xml). 

For consistency I would suggest to also include this element (including its *isExecuted* and *isSuccess* attributes) for files that *don't* result in any parse error or exception. Doing so would make it more straightforward to identify problematic/weird/damaged PDFs based on VeraPDF's output.

A quick search on "taskResult" in the issue tracker did turn up [this old issue](https://github.com/veraPDF/veraPDF-library/issues/1014), and the sample output posted there suggests that the *taskResult* result element was actually included by default in earlier VeraPDF versions.

I'm using veraPDF 1.22.3 on Linux Mint 20.1.

**Comment by bdoubrov** [comment]
Currently we add `<taskResult>` to the job only if there was a critical error that forced veraPDF to terminate the validation / feature extraction task. This can be used indeed as a detector for the corrupt PDF.

I don see however that the current schema is indeed not very consistent, and we'll see how we can improve it. One issue with always adding a `<taskResult>` to the job is that the job may include several tasks, such as parsing or validation. This is what is indicated in the property `type` of `<taskResult>`

**Comment by bitsgalore** [comment]
@bdoubrov Thanks for the clarification, wasn't aware of that. This makes me wonder whether having multiple *taskResult* elements per job (with different *type* values) would actually be a problem? Or perhaps it would create too much clutter?

**Comment by bdoubrov** [comment]
@bitsgalore We'd rather rename `<taskResult>` to `<taskException>` to make clear that the reason for this element is to report the ubnormal behavior rather than the task result. Would this make things more transparent?

**Comment by MaximPlusov** [comment]
Included into release 1.26
